presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp

```diff
@@ -207,26 +207,26 @@ std::string PrestoServerOperations::taskOperation(
     }
     case ServerOperation::Action::kListAll: {
       uint32_t limit;
+      const auto& limitStr = message->getQueryParam("limit");
       try {
-        const auto& limitStr = message->getQueryParam("limit");
         limit = limitStr == proxygen::empty_string
             ? std::numeric_limits<uint32_t>::max()
             : stoi(limitStr);
       } catch (std::exception& ex) {
-        VELOX_USER_FAIL(ex.what());
+        VELOX_USER_FAIL("Invalid limit provided '{}'.", limitStr);
       }
       std::stringstream oss;
       oss << "[";
       uint32_t count = 0;
-      for (const auto& task : taskMap) {
-        const auto& veloxTask = task.second->task;
+      for (auto taskItr = taskMap.begin(); taskItr != taskMap.end();
+           ++taskItr) {
+        const auto& veloxTask = taskItr->second->task;
         if (++count > limit) {
           oss << "... " << (taskMap.size() - limit) << " more tasks ...\n";
```

Comment:
@tanjialiang Would you double check that returned JSON is valid. Is seems problematic to add "... N more tasks ..." string in the middle of JSON array.